src/components/fields/Rating/Settings.tsx

```diff
@@ -1,48 +1,95 @@
 import { ISettingsProps } from "@src/components/fields/types";
-
-import { Slider, InputLabel, TextField, Grid } from "@mui/material";
+import RatingIcon from "@mui/icons-material/Star";
+import RatingOutlineIcon from "@mui/icons-material/StarBorder"
+import { InputLabel, TextField, Grid, FormControlLabel, Checkbox, Stack } from "@mui/material";
 import ToggleButton from "@mui/material/ToggleButton";
 import ToggleButtonGroup from "@mui/material/ToggleButtonGroup";
+import MuiRating from "@mui/material/Rating";
+import { get } from "lodash-es";
 
 export default function Settings({ onChange, config }: ISettingsProps) {
   return (
-    <>
-      <Grid container spacing={2} justifyItems="end" direction={"row"}>
-        <Grid item xs={6}>
-          <TextField
-            label="Maximum number of stars"
-            type={"number"}
-            value={config.max}
-            fullWidth
-            onChange={(e) => {
-              onChange("max")(parseInt(e.target.value));
-            }}
-            inputProps={{ min: 1, max: 20 }}
-          />
-        </Grid>
-        <Grid item xs={6}>
-          <InputLabel>Star fraction</InputLabel>
-          <ToggleButtonGroup
-            value={config.precision}
-            exclusive
-            fullWidth
-            onChange={(_, value) => {
-              onChange("precision")(value);
-            }}
-            aria-label="text alignment"
-          >
-            <ToggleButton value={0.25} aria-label="quarter">
-              1/4
-            </ToggleButton>
-            <ToggleButton value={0.5} aria-label="half">
-              1/2
-            </ToggleButton>
-            <ToggleButton value={1} aria-label="whole">
-              1
-            </ToggleButton>
-          </ToggleButtonGroup>
-        </Grid>
+    <Grid container spacing={2} justifyItems="end" direction={"row"}>
+      <Grid item xs={6}>
+        <TextField
+          label="Highest possible rating"
+          type={"number"}
+          value={config.max}
+          fullWidth
+          error={false}
+          onChange={(e) => {
+            let input = parseInt(e.target.value) || 0
+            if (input > 20) { input = 20 }
+            onChange("max")(input);
+          }}
```

Comment:
```suggestion
          onChange={(e) => {
            let input = parseInt(e.target.value) || 0
            if (input > 20) { input = 20 }
            onChange("max")(input);
          }}
          inputProps={{ min: 1, max: 20 }}
```
I’m not sure why the Rating field changes are here, but this line needs to be put back in. I can set the max to -1 or less and it causes an error.

Reply:
@mshamsrainey The rest looks good to me so I’ll commit this myself and merge now. Just wanted to make sure you see this